x-pack/plugins/security_solution/cypress/e2e/detection_rules/persistent_rules_table_state.cy.ts

```diff
@@ -0,0 +1,356 @@
+  const rule = getNewRule();
+
+  rule.name = name;
+  rule.tags = tags;
+  rule.enabled = enabled;
+
+  createCustomRule(rule, id);
+}
+
+function createTestRules(): void {
+  createRule('1', 'test 1', ['tag-a']);
+  createRule('2', 'rule 1', ['tag-b']);
+  createRule('3', 'rule 2', ['tag-b']);
+  createRule('4', 'rule 3', ['tag-b', 'tag-c']);
+  createRule('5', 'rule 4', ['tag-b']);
+  createRule('6', 'rule 5', ['tag-b', 'tag-c']);
+  createRule('7', 'rule 6', ['tag-b']);
+  createRule('8', 'rule 7', ['tag-b'], true);
+}
+
+function visitRulesTableWithState(urlTableState: Record<string, unknown>): void {
+  visit(SECURITY_DETECTIONS_RULES_MANAGEMENT_URL, { qs: { rulesTable: encode(urlTableState) } });
+}
+
+function setStorageState(storageTableState: Record<string, unknown>): void {
+  cy.window().then((win) => {
+    win.sessionStorage.setItem('securitySolution.rulesTable', JSON.stringify(storageTableState));
+  });
+}
+
+function changeRulesTableState(): void {
+  filterBySearchTerm('rule');
+  filterByTags(['tag-b']);
+  filterByCustomRules();
+  filterByDisabledRules();
+  sortByTableColumn('Rule', 'asc');
+  setRowsPerPageTo(5);
+}
+
+function expectRulesTableState(): void {
+  expectFilterSearchTerm('rule');
+  expectFilterByTags(['tag-b']);
+  expectFilterByCustomRules();
+  expectFilterByDisabledRules();
+  expectTableSorting('Rule', 'asc');
+  expectRowsPerPage(5);
+}
+
+function expectDefaultRulesTableState(): void {
+  expectFilterSearchTerm('');
+  expectNoFilterByTags();
+  expectNoFilterByElasticOrCustomRules();
+  expectNoFilterByEnabledOrDisabledRules();
+  expectTableSorting('Enabled', 'desc');
+  expectRowsPerPage(20);
+  expectTablePage(1);
+}
+
+function expectManagementTableRules(ruleNames: string[]): void {
+  expectNumberOfRules(RULES_MANAGEMENT_TABLE, ruleNames.length);
+
+  for (const ruleName of ruleNames) {
+    expectToContainRule(RULES_MANAGEMENT_TABLE, ruleName);
+  }
+}
+
+describe('Persistent rules table state', () => {
+  before(() => {
+    cleanKibana();
+    createTestRules();
+    login();
+  });
+
+  beforeEach(() => {
+    resetRulesTableState();
+  });
+
+  describe('while on a happy path', () => {
+    it('activates management tab by default', () => {
+      visit(SECURITY_DETECTIONS_RULES_URL);
+
+      expectRulesManagementTab();
+    });
+
+    it('prefers url state over storage state', () => {
+      setStorageState({
+        searchTerm: 'test',
+        tags: ['tag-c'],
+        source: 'prebuilt',
+        enabled: true,
+        field: 'severity',
+        order: 'desc',
+        perPage: 10,
+      });
+
+      visitRulesTableWithState({
+        searchTerm: 'rule',
+        tags: ['tag-b'],
+        source: 'custom',
+        enabled: false,
+        field: 'name',
+        order: 'asc',
+        perPage: 5,
+        page: 2,
+      });
+
+      expectRulesManagementTab();
+      expectRulesTableState();
+      expectTablePage(2);
+      expectManagementTableRules(['rule 6']);
```

Comment:
Following up on our discussion here: https://github.com/elastic/kibana/pull/149638#discussion_r1106885838. I think it would be helpful to create a separate test case that checks whether the sorting, filtering, and pagination from the URL state are correctly applied to the table rules. This test case can specifically verify that the correct rules are displayed on the rules table. Additionally, I propose we remove the checks that verify rules on the table from all other cases where it's not necessary. This would help to keep our test cases minimal and more resilient to future changes.